packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -545,6 +545,61 @@ export function findRootMergeBlock(
 	return maybeRoot?.mergeTree !== undefined ? maybeRoot : undefined;
 }
 
+/**
+ * @param segment - The segment to slide from.
+ * @param cache - Optional cache mapping segments to their sliding destinations.
+ * Excursions will be avoided for segments in the cache, and the cache will be populated with
+ * entries for all segments visited during excursion.
+ * This can reduce the number of times the tree needs to be scanned if a range containing many
+ * SlideOnRemove references is removed.
+ * @returns The segment a SlideOnRemove reference should slide to, or undefined if there is no
+ * valid segment (i.e. the tree is empty).
+ * @internal
+ */
+export function getSlideToSegment(
```

Comment:
client.getSlideToSegment does some extra stuff, in particular handling the offset for slide direction. This results in divergence between uses of the two, which might be not desirable or intentional. (esp. the way it's used in revertibles) Can we just make this match client and then deprecate the client implementation? (And update usages)